src/gWhisperSetting/GWhisperSetting.cpp

```diff
@@ -0,0 +1,146 @@
+
+#include "GWhisperSetting.hpp"
+#include "libArgParse/ArgParse.hpp"
+
+using json = nlohmann::json;
+
+void gWhisperSetting::parseConfigFile(const std::string &f_inputFile){
+    // Parse JSON File into member SON object
+    std::ifstream ifs(f_inputFile.c_str());
+
+    if (!ifs.is_open())
+    {
+        std::cout << "Error while opening config file at: " << f_inputFile << std::endl;
+        exit(EXIT_FAILURE);
+    }
+    // TODO: check for empty value with  nlohmann::isempty()
+    ifs >> m_config;
+    ifs.close();
+
+}
+
+void gWhisperSetting::retrieveConfigParameters(json &f_startElement){
+    //recursively go over config file and retrieve all parameter keys
+    if(f_startElement.is_structured())
+    {
+         for (const auto &item : f_startElement.items())
+        {
+            m_configParameters.push_back(item.key());
+
+            if (item.value().is_object())
+            {
+                retrieveConfigParameters(item.value());
+            }
+        }
+    }
+}
+
+json gWhisperSetting::findParameterSettingInConfig(const std::string &f_parameter,const  json &f_startElement)
+{
+    // TODO: If enough time, optimmize for
+    bool containsParameter = false;
+    json setting;
+
+    for (const auto& item : f_startElement.items())
+    {
+        if (item.value().contains(f_parameter))
+        {
+            //Abbruchkrit
+
+            setting[f_parameter] = item.value().at(f_parameter);
+            break;
+        }
+
+        for(auto &innerElement : item.value().items())
+        {   
+            if (innerElement.key() == "")
+            {
+                break; 
+            }
+
+            // Only Call recursion, if current element contains further elements
+            if(innerElement.value().is_structured())
+            {
+                json tempElement;
+                tempElement[innerElement.key()] = innerElement.value();
+                return findParameterSettingInConfig(f_parameter, tempElement);
+            }
+            else
+            {
+                continue;
+            }
+        }
+    }
+    return setting;
+}
+
+// f_paramaetr entrpricht label des Grammatik Elements in ParseTree
+std::string gWhisperSetting::lookUpSetting(const std::string &f_parameter, ArgParse::ParsedElement &f_parseTree)
+{
+    bool containsParameter;
+    std::string setting;
+    json someJson;
+
+    if (f_parseTree.findFirstChild(f_parameter) != "")
+    {
+        setting = f_parseTree.findFirstChild(f_parameter);
+    }
+    else
+    {
+        someJson = findParameterSettingInConfig(f_parameter, m_config); //copy
+        if (!someJson.is_null())
+        {    
+            if (!someJson.at(f_parameter).is_null())
+            {
+                containsParameter = true;
+                setting = someJson.at(f_parameter);
+            }
+
+            if(someJson.at(f_parameter).is_null())
+            {
+                setting = ""; // TODO: Is this the right semantic for Timeout?
```

Comment:
I think we can use empty string to indicate not found here (we do this with parse tree too). However this is not a timeout xD